SOD.Common/Helpers/PluginDetection.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using BepInEx;
+using BepInEx.Configuration;
+using BepInEx.Unity.IL2CPP;
+using UniverseLib;
+
+namespace SOD.Common.Helpers
+{
+    public sealed class PluginDetection
+    {
+        /// <summary>
+        /// Raised right after the BepInEx chainloader finishes loading the
+        /// last plugin.
+        /// </summary>
+        public event EventHandler OnAllPluginsFinishedLoading;
+
+        /// <summary>
+        /// True if the BepInEx chainloader has finished loading all plugins,
+        /// false otherwise (is set to true after OnAllPluginsFinishedLoading).
+        /// </summary>
+        public bool AllPluginsFinishedLoading { get; private set; } = false;
+
+        private bool _initialized = false;
+
+        internal PluginDetection()
+        {
+            Initialize();
+        }
+
+        internal void Initialize()
```

Comment:
Make this method private, and I think that makes the _initialized field redundant

Reply:
agreed, will change

Reply:
You can probably even just move the whole logic into the constructor :P